### DISCUSSION TITLE
Co dělat, když chci být playtester

### DISCUSSION THREAD

**Discussion by Tab-e-n** [Tab-e-n/blue-runner#61] [Announcements]
# Co dělat, když chci být playtester

SR je technicky hra v předběžném přístupu, takže potřebuje někoho, aby ji playtestoval. Tohle bývalo děláno tak, že jsem vyexportoval SR a poslal jej k ostatním, co o to měli zájem, ale po minulém katastrofě co byla 1.0.0 jsem se rozhodl to začít dělat trochu jinak. Chtěl bych aby kdokoliv, co se chce připojit jako playtester, mohl hrát nejnovější verzi SR bez toho, aby musel čekat, až mu pošlu vyexportovaný soubor té verze, ale mohl to hrát kdy se mu zachce.

V tomto návodě vám chci teda popsat, jak tohoto můžete dosáhnout. Samozřejmě mnoha z vás už bude vědět, jak to udělat, a i když je tenhle návod spíše určený pro člověka, co vůbec nic neví, doporučuji vám si ho také přečíst.

### Předpoklady / Co potřebujete

Budete potřebovat si stáhnout aplikace [git](https://git-scm.com/downloads) a [Godot 3.2.3](https://downloads.tuxfamily.org/godotengine/3.2.3/), a také potřebujete dostatek místa na disku.

## Stáhnutí zdrojového kódu

Začneme s stáhnutím zdrojového kódu. Prvně si najděte na svém počítači místo, kde chcete SR mít. Nemusíte dělat novou složku přímo pro SR, protože se při tomto návodě udělá sama. Nezáleží na tom, jak se místo jmenuje, nebo kde bude. Poté, co si vyberete místo, spusťte příkazový řádek (command promt). Pomocí příkazu cd se přesuňte do té složky, kam chcete SR dát. Pokud složka není na disku C, budete muset před cd dát ještě jméno disku. Příkazy budou vypadat nějak takhle:

```
D:
cd "Hry\Sonic Runner"
```
_v tomto případě skončíte v D:\Hry\Sonic Runner\. Pokud má složka v názvu speciální znaky nebo mezery, musí být v ""._

Až se donaviguje do své složky, tak můžeme stáhnout zdrojový kód. Naklonujte SR repozitář:

```
git clone https://github.com/Tab-e-n/blue-runner
```

Ve místě, kde jste chtěli SR dát, by se měla objevit složka "blue-runner", což znamená, že máte zdrojový kód stáhnutý. Pokud se vám přestane líbit lokace, kde jste SR uložili, klidně můžete SR přesunout.

Jako poslední věc, co bych doporučoval udělat, je malý prográmek, který bude aktualizovat hru. Vedle té nové složky udělejte nový textový soubor (.txt). Do něho zapište následující příkazy:

```
cd blue-runner
git pull
pause
```
_Pokud jste přejmenovali složku blue-runner na něco jiného, zadejte místo "cd blue-runner" příkaz "cd [vaše jméno složky]"._

Poté co to napíšete, zmáčkněte čudlík "Uložit jako", a uložte program né jako .txt, ale jako .bat. Pokud se vám program uložil správně, měli by jste mýt dva soubory vedle složky: 

![Snímek obrazovky 2023-01-29 120628](https://user-images.githubusercontent.com/100419380/215322028-0d5b9067-1625-40ae-af6e-ebe7478d00c0.png)

Pokud soubor .bat vypadá jak soubor na obrázku, tak jste s tímto krokem hotový. Soubor .txt můžete smazat, už ho nepotřebujete. Teďka vždy, když chcete blue update, jenom spusťte tento program a on vám ho sám stáhne.

### Co jsem vůbec stáhl?

Repozitář blue-runner zahrnuje:

1. Hru Sonic Runner (pod složkou blue-runner)
2. Sonic Runner Editor (pod složkou blue-editor)
3. Příkladový mod (pod složkou default-mod)
4. Všechny textury SR a SR Editoru (pod složkami Visual a gimp)
5. Todo Listina (todo.txt)
6. Patch notes (update_notes.txt)

Ostatní soubory nejsou pro vás moc relevantní

## Importování SR do Godotu

Tohle je poslední věc, co potřebujete udělat, než budete moci hrát hru. Zapněte [Godot 3.2.3](https://downloads.tuxfamily.org/godotengine/3.2.3/). Měl by na vás vyskočit Project Manager:

![editor_ui_intro_project_manager_01](https://user-images.githubusercontent.com/100419380/215323200-1f216844-68d5-44e8-9a43-211ec60e7a85.png)

V Project Manageru klikněte na čudlík "Import", mělo by vyskočit tohle:

![import](https://user-images.githubusercontent.com/100419380/215323287-5d01f0fd-fed7-4483-8f9a-f3f574ad4883.png)

Zmáčkněte browse a bežte do složky, kde je SR umístěno, a najděte soubor project.godot. SR a SR Editor oba mají vlastní project.godot, musíte je proto importovat zvlášť. Poté co najdete soubor project.godot, zmáčkněte "Import & Edit". To vám project otevře v novém okně a zavře Project Manager. Poté můžete kliknout na šipku v pravém horním rohu nebo zmáčknout F5 aby jste projekt spustily.

## Instrukce poté, co jste dokončili instalaci

Teď už můžete začít hrát hru. Všechny věci, co od playtesterů potřebuji aby udělali, jsou napsané v README.md pod sekcí "Mohu nějak pomoci s vývojem hry?".

Předem vám děkuji za playtestování mé hry!

**Comment by Olda-Hal** [comment]
****jupí, vůbec to nechápu 🥺😥
